drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/main/java/org/drools/workbench/screens/guided/rule/client/editor/AttributeSelectorPopup.java

```diff
@@ -16,81 +16,63 @@
 import org.uberfire.ext.widgets.common.client.common.popups.FormStylePopup;
 
-public class AttributeSelectorPopup extends FormStylePopup {
+public abstract class AttributeSelectorPopup extends FormStylePopup {
 
-    private final TextBox box = new TextBox();
+    protected TextBox box;
+    protected ListBox list;
 
-    public AttributeSelectorPopup( final RuleModel model,
-                                   boolean lockLHS,
-                                   boolean lockRHS,
-                                   final Command refresh ) {
-        super( getImage(),
+    public AttributeSelectorPopup() {
+        super( GuidedRuleEditorImages508.INSTANCE.Configure(),
                GuidedRuleEditorResources.CONSTANTS.AddAnOptionToTheRule() );
+    }
 
-        setTextBox( model,
-                    refresh );
-
-        setListBox( model,
-                    refresh );
-
-        setFreezePanel( model,
-                        lockLHS,
-                        lockRHS,
-                        refresh );
-
+    protected final void initialize() {
+        initialize( new TextBox(),
+                    new ListBox() );
     }
 
-    private static Image getImage() {
-        Image image = new Image( GuidedRuleEditorResources.INSTANCE.images().config() );
-        image.setAltText( GuidedRuleEditorResources.CONSTANTS.Config() );
-        return image;
+    protected final void initialize( TextBox box, ListBox list ) {
+        this.box = box;
+        this.list = list;
+        setMetadataPanel();
+        setAtributesPanel();
     }
 
-    private void setTextBox( final RuleModel model,
-                             final Command refresh ) {
-        ( (InputElement) box.getElement().cast() ).setSize( 15 );
+    private void setMetadataPanel() {
+        box.getElement().setAttribute( "size",
+                                       "15" );
 
         DirtyableHorizontalPane horiz = new DirtyableHorizontalPane();
         horiz.add( box );
-        horiz.add( getAddButton( model,
-                                 refresh,
-                                 box ) );
+        horiz.add( getAddButton() );
 
         addAttribute( GuidedRuleEditorResources.CONSTANTS.Metadata3(),
                       horiz );
 
     }
 
-    private void setListBox( final RuleModel model,
-                             final Command refresh ) {
-
-        final ListBox list = RuleAttributeWidget.getAttributeList();
+    private void setAtributesPanel() {
+        for ( String item : getAttributes() ) {
+            list.addItem( item );
+        }
 
         // Remove any attributes already added
-        for ( RuleAttribute at : model.attributes ) {
+        for ( String at : getDuplicates() ) {
             for ( int iItem = 0; iItem < list.getItemCount(); iItem++ ) {
-                if ( list.getItemText( iItem ).equals( at.getAttributeName() ) ) {
+                if ( list.getItemText( iItem ).equals( at ) ) {
```

Comment:
Why check if the attribute names are equal before removing, when the method you called is ```getDuplicates()``` which (in my mind) suggests the collection already contains the attributes to remove?